src/dbnode/storage/series/series.go

```diff
@@ -448,26 +450,33 @@ func (s *dbSeries) bootstrap(
 		return result, nil
 	}
 
-	s.loadWithLock(bootstrappedBlocks, blockStates)
+	s.loadWithLock(true, bootstrappedBlocks, blockStates)
 	result.NumBlocksMovedToBuffer += int64(bootstrappedBlocks.Len())
 
 	return result, nil
 }
 
-func (s *dbSeries) load(
-	bootstrappedBlocks block.DatabaseSeriesBlocks,
-	blockStates BootstrappedBlockStateSnapshot,
-) {
-	s.Lock()
-	s.loadWithLock(bootstrappedBlocks, blockStates)
-	s.Unlock()
-}
-
 func (s *dbSeries) loadWithLock(
-	bootstrappedBlocks block.DatabaseSeriesBlocks,
+	isBootstrap bool,
+	blocksToLoad block.DatabaseSeriesBlocks,
 	blockStates BootstrappedBlockStateSnapshot,
 ) {
-	for _, block := range bootstrappedBlocks.AllBlocks() {
+	for _, block := range blocksToLoad.AllBlocks() {
```

Comment:
As mentioned offline, this is fine now, but in the future we should measure the performance of cold flushes when there is no cold data. The `ColdFlushEnabled` flag should probably just control whether cold writes are accepted or not - other functions like bootstrapping and flushes should be free to use the cold flush mechanism regardless.

Reply:
agreed, maybe lets add it as a cold writes follow up issue or track it as part of plan to combine warm flush and cold flush into one process?